File: HospitalApp/src/main/java/viewmodel/panes/QRCodePane.java
package viewmodel.panes;

import javafx.geometry.Pos;
import javafx.scene.control.Button;
import javafx.scene.image.ImageView;
import javafx.scene.layout.BorderPane;
import main.Helper;

public class QRCodePane extends BorderPane {
    private ImageView qrView;
    private Button backButton;

    public QRCodePane() {
        qrView = new ImageView();
        backButton = new Button("Back");

        connectComponents();
        styleComponents();
        setCallbacks();
    }

    public void drawQRCode(String str) {
        Helper.getInstance().drawQRCode(qrView, str);
    }

    private void connectComponents() {
        setAlignment(qrView, Pos.CENTER);
        setAlignment(backButton, Pos.CENTER);
        setCenter(qrView);
        setBottom(backButton);
    }

    private void styleComponents() {

    }

    private void setCallbacks() {

    }
}

File: ValidatorApp/src/main/kotlin/viewmodel/panes/QRCodePane.kt
package viewmodel.panes

import javafx.geometry.Pos
import javafx.scene.control.Button
import javafx.scene.image.ImageView
import javafx.scene.layout.BorderPane
import main.Helper

class QRCodePane : BorderPane() {
    private val qrView: ImageView
    private val backButton: Button

    init {
        qrView = ImageView()
        backButton = Button("Back")

        connectComponents()
        styleComponents()
        setCallbacks()
    }

    fun drawQRCode(str: String) {
        Helper.drawQRCode(qrView, str)
    }

    private fun connectComponents() {
        BorderPane.setAlignment(qrView, Pos.CENTER)
        BorderPane.setAlignment(backButton, Pos.CENTER)
        center = qrView
        bottom = backButton
    }

    private fun styleComponents() {

    }

    private fun setCallbacks() {

    }
}
